src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage/segment_meta_cstore.h"
+
+namespace cloud_storage {
+
+class segment_meta_cstore::impl {
+public:
+private:
+};
+
+segment_meta_cstore::const_iterator segment_meta_cstore::begin() {
+    throw "not implemented";
+}
```

Comment:
pure virtual?

Reply:
what i meant was: it seems like this is defining an interface, so make it an interface